docs/guides/api-gateway/kubernetes.mdx

```diff
@@ -0,0 +1,414 @@
+        certificateRefs:
+          - name: ngrokcert
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: ngrok-route
+  namespace: prod
+spec:
+  parentRefs:
+    - kind: Gateway
+      name: ngrok-gateway
+  hostnames:
+    #highlight-start
+    - "{YOUR_NGROK_DOMAIN}"
+    #highlight-end
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /
+      backendRefs:
+        #highlight-start
+        - name: { SERVICE_NAME }
+          port: { PORT }
+          #highlight-end
+          kind: Service
+```
+
+Write the manifest to disk and apply to your cluster.
+
+```bash
+kubectl apply -f ngrok-api-gateway.yaml
+```
+
+With the ngrok Kubernetes Operator deployed and configured as an API gateway,
+your API service is now publicly available at `{YOUR_NGROK_DOMAIN}`. You can now
+test your API gateway and service.
+
+```bash
+curl https://{YOUR_NGROK_DOMAIN}
+
+{"message":"Pong from ngrok!","host":"{YOUR_NGROK_DOMAIN}","client_ip":"{YOUR_IP_ADDRESS}","method":"GET","url":"/","time":"2024-10-16T13:51:37.836Z","status":200}
+```
+
+## **Step 3**. Set up Traffic Policy to shape traffic into/out of your API service {#set-up-traffic-policy}
+
+ngrok's [Traffic Policy](/docs/http/traffic-policy/index.mdx) module allows you
+to conditionally act upon traffic as it arrives and departs your API gateway. In
+a Kubernetes context, Traffic Policy builds on top of the existing YAML
+manifests you already use to declare your deployments and services.
+
+When the ngrok Kubernetes Operator initiates in your cluster, and your
+`HTTPRoute` configuration contains one or more `filter`
+dictionaries that reference the `NgrokTrafficPolicy` CRD, the Operator pipes the
+raw YAML to ngrok's network to provision the action and enforce it upon incoming
+and outgoing traffic.
+
+Rate limiting&mdash;one of the most in-demand API gateway features&mdash;
+is both quick to implement and highly flexible based on your use case.
+
+### Define a rate limiting rule
+
+Create a new YAML file named `ngrok-rate-limit.yaml` with the following content:
+
+```yaml
+---
+kind: NgrokTrafficPolicy
+apiVersion: ngrok.k8s.ngrok.com/v1alpha1
+metadata:
+  name: rate-limiting
+  namespace: prod
+spec:
+  policy:
+    inbound:
+      - name: "Rate limit POST requests"
+        expressions:
+          - "req.method == 'POST' || req.method == 'PUT'"
+        actions:
+          - type: "rate-limit"
+            config:
+              name: "Only allow 10 requests per minute"
+              algorithm: "sliding_window"
+              capacity: 10
+              rate: "60s"
+              bucket_key:
+                - "conn.client_ip"
+```
+
+This CRD instantiates a rate limiting rule with the following properties:
+
+- Via the `expression`, it only applies to requests with the `POST` or `PUT`
+  HTTP method&mdash;all others, like `GET`, are unaffected.
+- Allows for a maximum of `10` requests in a given `60s` window, both of which
```

Comment:
This one is tricky. The rolling algorithm allows 10 requests "over the last 60 seconds" rather than in a give 60 second window. It's a technical nuance that Natasha was adamant about when I reviewed [her blog post](https://ngrok.com/blog-post/new-feature-rate-limiting) but the technically accuracy may be overkill here. I'm including this info for consideration.

Reply:
Thanks! That's great feedback.